lib/tools/apk-utils.js

```diff
@@ -359,7 +360,8 @@ apkUtilsMethods.install = async function (apk, options = {}) {
   }
 
   const executeInstall = async (args) => {
-    const output = await this.adbExec(['install', ...args, apk], {timeout});
+    args = args.map((arg) => quote([arg]));
+    const output = await this.adbExec(['install', ...args, quote([apk])], {timeout});
```

Comment:
was this piece of code also problematic with spaces?

Reply:
apk path here also must be quoted (since ui-automator server APK path also contains space char if Appium server contains space char).
But quotation for other args are redundant, so I will remove it.

Reply:
I finally found this fix quotation is unnecessary (sorry!), and removed it.